src/game/client/heightmap.cpp

```diff
@@ -658,250 +658,268 @@ int Invert_Value(int value)
     }
 }
 
+/**
+ * Update the center of the HeightMapRenderObjClass.
+ *
+ * @param camera   Pointer to the CameraClass object.
+ * @param lights   Pointer to the RefMultiListIterator<RenderObjClass> object.
+ */
 void HeightMapRenderObjClass::Update_Center(CameraClass *camera, RefMultiListIterator<RenderObjClass> *lights)
 {
+    // Check if the height map, updating flag, and vertex buffer are valid
     if (m_map != nullptr && !m_updating && m_vertexBufferTiles != nullptr) {
+        // Update the center of the base height map render object
         BaseHeightMapRenderObjClass::Update_Center(camera, lights);
         m_updating = true;
 
+        // Perform a full update if necessary
         if (m_needFullUpdate) {
             m_needFullUpdate = false;
             Update_Block(0, 0, m_x - 1, m_y - 1, m_map, lights);
             m_updating = false;
             return;
         }
 
+        // Check if the dimensions of the height map have been reached
         if (m_x >= m_map->Get_X_Extent() && m_y >= m_map->Get_Y_Extent()) {
             m_updating = false;
             return;
         }
 
-        int i1 = 1;
+        // Initialize variables
+        int increment = 1;
         Matrix3D camera_transform = camera->Get_Transform();
         Vector3 camera_pos = camera->Get_Position();
-        Vector3 v1;
-        Vector3 v2;
+        Vector3 view_direction;
+        Vector3 intersection_point;
         const ViewportClass &viewport = camera->Get_Viewport();
-        int height = WorldHeightMap::Get_Max_Height_Value();
+        int max_height = WorldHeightMap::Get_Max_Height_Value();
 
+        // Find the minimum display height in the height map
         for (int x = 0; x < m_x; x++) {
             for (int y = 0; y < m_y; y++) {
                 unsigned char display_height = m_map->Get_Display_Height(x, y);
 
-                if (display_height < height) {
-                    height = display_height;
+                if (display_height < max_height) {
+                    max_height = display_height;
                 }
             }
         }
 
-        float z = height;
-        Vector2 min;
-        Vector2 max;
-        camera->Get_View_Plane(min, max);
-        float x_dist = max.X - min.X;
-        float y_dist = max.Y - min.Y;
-        float f1 = -1.0f;
-        float f2 = 200000.0f;
-        float f3 = -200000.0f;
-        float f4 = 200000.0f;
-        float f5 = f3;
-
+        float z = max_height;
+        Vector2 viewport_min;
+        Vector2 viewport_max;
+        camera->Get_View_Plane(viewport_min, viewport_max);
+        float x_dist = viewport_max.X - viewport_min.X;
+        float y_dist = viewport_max.Y - viewport_min.Y;
+        float scale_factor = -1.0f;
+        float min_x = 200000.0f;
```

Comment:
Right now there is
```
x_dist
min_x
```
among others.

Perhaps it makes sense to name those x,y variables in consistent order?

Perhaps x,y always at the end of the name?